torch/utils/data/datapipes/utils/snapshot.py

```diff
@@ -0,0 +1,48 @@
+from torch.utils.data.datapipes.datapipe import IterDataPipe
+from torch.utils.data.graph_settings import apply_shuffle_seed
+
+
+# TODO: Caveats
+#   1. Caller (either the ReadingService or DataLoader) must pass in the initial RNG
+#   2. `in_batch_shuffle` and `bucketbatch` are not compatible with this because they currently
+#      lack the option to `set_seed`.
+
+def _simple_snapshot_graph(datapipe: IterDataPipe, n_iterations: int, rng=None) -> None:
```

Comment:
The general idea of the simple snapshot is that we will:
1. Create a new iterator
2. Move that iterator forward by `n_iterations`
3. Save that iterator to the DataPipe
4. The next time `iter` is called on the DataPipe, use the forwarded iterator

Reply:
What would be the expected behavior if we:
1. Create a iterator
2. load snapshot
3. continue iterate over the iterator

Reply:
Creating an iterator will set `datapipe._restored = True`, and you can't load snapshot after, because of the condition here:
https://github.com/pytorch/pytorch/blob/7a74a5c1d91c4c4c968e0953dc5a0cc7520be48f/torch/utils/data/datapipes/utils/snapshot.py#L27-L30

Let me add a warning there (let me know if you prefer it to be an exception).

Reply:
If that's the case, I think raising an Error is a more proper solution.